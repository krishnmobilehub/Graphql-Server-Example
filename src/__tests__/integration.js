const { createTestClient } = require('apollo-server-testing');
const gql = require('graphql-tag');

const { constructTestServer } = require('./__utils');

const { mockStore } = require('../datasources/__tests__/user');

const CHECK_USER_NAME = gql`
    query UsernameAvailable($username: String!) {
        isUsernameAvailable(username: $username)
    }
`;

const CREATE_USER = gql`
    mutation CreateUser($username: String!, $password: String!, $phoneNumber: String!, $verificationCode: String!) {
       createUser(username: $username, password: $password,phoneNumber: $phoneNumber,verificationCode: $verificationCode)
    }
`;

describe('Queries', () => {
    it('fetches user from username', async () => {
        const { server, userAPI } = constructTestServer({
            context: () => ({ user: { id: 1, username: 'test', password:'test@123', phoneNumber:'9876543210',verificationCode:'2344' } }),
          });

          userAPI.store = mockStore;

          const { query } = createTestClient(server);
          const res = await query({ query: CHECK_USER_NAME, variables: { username: "test" } });
          expect(res.data.isUsernameAvailable) == true;
    })
});

describe('Mutations', () => {
    it('returns message on create user', async () => {
      const { server, userAPI } = constructTestServer({
        context: () => {},
      });
  
      userAPI.store = mockStore;
    
      const { mutate } = createTestClient(server);
      const res = await mutate({
        mutation: CREATE_USER,
        variables: { username: 'test1', password:'test@1234', phoneNumber:'9876344543210',verificationCode:'2344' },
      });
      expect(res).toMatchSnapshot();
    });
});
