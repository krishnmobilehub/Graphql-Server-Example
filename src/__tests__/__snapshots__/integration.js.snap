// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[UserAPI.findUserByUserName] returns null for invalid username 1`] = `null`;

exports[`[UserAPI.getUserById] looks up user by userId 1`] = `null`;

exports[`[UserAPI.getUserById] returns empty array if nothing found 1`] = `null`;

exports[`Mutations returns message on create user 1`] = `
Object {
  "data": Object {
    "createUser": "Failed",
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries fetches user from username 1`] = `
Object {
  "data": Object {
    "isUsernameAvailable": null,
  },
  "errors": Array [
    [GraphQLError: this.store.users.find is not a function],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
